File: src/main/java/com/refinedmods/refinedstorage/container/StorageContainer.java
package com.refinedmods.refinedstorage.container;

import com.refinedmods.refinedstorage.RSContainers;
import com.refinedmods.refinedstorage.container.slot.filter.FilterSlot;
import com.refinedmods.refinedstorage.blockentity.StorageBlockEntity;
import net.minecraft.world.entity.player.Player;

public class StorageContainer extends BaseContainer {
    public StorageContainer(StorageBlockEntity storage, Player player, int windowId) {
        super(RSContainers.STORAGE_BLOCK, storage, player, windowId);

        for (int i = 0; i < 9; ++i) {
            addSlot(new FilterSlot(storage.getNode().getFilters(), i, 8 + (18 * i), 20));
        }

        addPlayerInventory(8, 141);

        transferManager.addItemFilterTransfer(player.getInventory(), storage.getNode().getFilters());
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/container/StorageContainer.kt
package com.refinedmods.refinedstorage.container

import com.refinedmods.refinedstorage.RSContainers
import com.refinedmods.refinedstorage.api.network.grid.IGrid.slotId
import com.refinedmods.refinedstorage.container.slot.filter.FilterSlot
import com.refinedmods.refinedstorage.tile.StorageTile
import net.minecraft.entity.player.PlayerEntity

class StorageContainer(storage: StorageTile, player: PlayerEntity, windowId: Int) : BaseContainer(RSContainers.STORAGE_BLOCK, storage, player, windowId) {
    init {
        for (i in 0..8) {
            addSlot(FilterSlot(storage.getNode().filters, i, 8 + 18 * i, 20))
        }
        addPlayerInventory(8, 141)
        transferManager.addItemFilterTransfer(player.inventory, storage.getNode().filters)
    }
}